sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/internal/view/AttributesProcessor.java

```diff
@@ -0,0 +1,198 @@
+          result.putAll(incoming);
+          return result.build();
+        });
+  }
+
+  /**
+   * Creates a processor which appends (exactly) the given attributes.
+   *
+   * <p>These attributes will not override those attributes provided by instrumentation.
+   *
+   * @param attributes Attributes to append to measurements.
+   */
+  public static AttributesProcessor append(Attributes attributes) {
+    return simple(incoming -> attributes.toBuilder().putAll(incoming).build());
+  }
+
+  /** Creates a simple attributes processor with no access to context. */
+  static AttributesProcessor simple(UnaryOperator<Attributes> processor) {
+    return new AttributesProcessor() {
+
+      @Override
+      public Attributes process(Attributes incoming, Context context) {
+        return processor.apply(incoming);
+      }
+
+      @Override
+      public boolean usesContext() {
+        return false;
+      }
+    };
+  }
+
+  /** Creates an Attributes processor that has access to baggage. */
+  static AttributesProcessor onBaggage(BiFunction<Attributes, Baggage, Attributes> processor) {
+    return new AttributesProcessor() {
+      @Override
+      public Attributes process(Attributes incoming, Context context) {
+        return processor.apply(incoming, Baggage.fromContext(context));
+      }
+
+      @Override
+      public boolean usesContext() {
+        return true;
+      }
+    };
+  }
+
+  static final AttributesProcessor NOOP = simple(incoming -> incoming);
+
+  /** A {@link AttributesProcessor} that runs a sequence of processors. */
+  @Immutable
+  static final class JoinedAttribtuesProcessor extends AttributesProcessor {
+    private final Collection<AttributesProcessor> processors;
+    private final boolean usesContextCache;
+
+    JoinedAttribtuesProcessor(Collection<AttributesProcessor> processors) {
+      this.processors = processors;
+      this.usesContextCache =
+          processors.stream().map(AttributesProcessor::usesContext).reduce(false, (l, r) -> l || r);
+    }
+
+    @Override
+    public Attributes process(Attributes incoming, Context context) {
+      Attributes result = incoming;
+      for (AttributesProcessor processor : processors) {
+        result = processor.process(result, context);
+      }
+      return result;
+    }
+
+    @Override
+    public boolean usesContext() {
+      return usesContextCache;
+    }
+
+    @Override
+    public AttributesProcessor then(AttributesProcessor other) {
+      ArrayList<AttributesProcessor> newList = new ArrayList<>(processors);
+      if (other instanceof JoinedAttribtuesProcessor) {
+        newList.addAll(((JoinedAttribtuesProcessor) other).processors);
+      } else {
+        newList.add(other);
+      }
+      return new JoinedAttribtuesProcessor(newList);
+    }
+
+    AttributesProcessor prepend(AttributesProcessor other) {
+      ArrayList<AttributesProcessor> newList = new ArrayList<>(processors.size() + 1);
```

Comment:
#nit: ArrayList is irrelevant. Same for line 182. 
```suggestion
      List<AttributesProcessor> newList = new ArrayList<>(processors.size() + 1);
```